### PR TITLE
Add Sparse support for torch.sqrt

### PR DESCRIPTION
**Pull request by hlahkar**
Fixes #{issue number}

### CODE REVIEW

**Comment by ezyang** [comment]
@pearu, over to you

**Review by aocsa — COMMENTED** [review]
Thanks @hlahkar  for bringing here `sqrt` for sparse tensors. I suggested a couple of changes. Besides that, It seems ready as the tests are generated using `common_methods_invocations.py`.

File: aten/src/ATen/native/sparse/SparseTensorMath.cpp [hunk 1]
```diff
@@ -166,6 +166,34 @@ SparseTensor& asin_sparse_(SparseTensor& t) {
...
 
+// --------------------------------------------------------------------
+// sqrt(SparseTensor)
+// --------------------------------------------------------------------
+
+// TODO: add in-place variant
+
+SparseTensor& sqrt_out_sparse(SparseTensor& r, const SparseTensor& t_) {
+  AT_ASSERT(r.is_sparse());
+  AT_ASSERT(t_.is_sparse());
```
[the remark this answers is not in the document]
Comment (hlahkar) [reply aten/src/ATen/native/sparse/SparseTensorMath.cpp]: @aocsa - Thanks for the comments. Have updated the PR

**Review by aocsa — APPROVED** [review]

File: aten/src/ATen/native/sparse/SparseTensorMath.cpp [hunk 2]
```diff
@@ -166,6 +166,34 @@ SparseTensor& asin_sparse_(SparseTensor& t) {
...
+// --------------------------------------------------------------------
+
+// TODO: add in-place variant
+
+SparseTensor& sqrt_out_sparse(SparseTensor& r, const SparseTensor& t_) {
+  TORCH_CHECK(r.is_sparse(), "Tensor should be sparse");
+  TORCH_CHECK(t_.is_sparse(), "Tensor should be sparse");
+
+  // This coalesce is why we can't easily provide an inplace variant
+  SparseTensor t = t_.coalesce();
```
Comment (aocsa) [inline aten/src/ATen/native/sparse/SparseTensorMath.cpp:180]: For this case I would suggest the following condition: 
```

if (is_same_tensor(r, t)) {
    TORCH_CHECK(
        r.is_coalesced(),
        "sqrt: in-place on uncoalesced tensors is not supported");
  } 
```

**Pull request closed without merging**